parsons/scytl/scytl.py

```diff
@@ -0,0 +1,561 @@
+import urllib
+import zipfile
+import csv
+import requests
+import urllib.error
+import urllib.request
+import xml.etree.ElementTree as ET
+from datetime import datetime
+from dateutil.parser import parse as parsedate
+from pytz import timezone
+from io import BytesIO, StringIO
+
+clarity_url = 'https://results.enr.clarityelections.com/'
+
+get_version_url = \
+    clarity_url + '{administrator}/{election_id}/current_ver.txt'
+summary_csv_url_template = \
+    clarity_url + '{administrator}//{election_id}/{version_num}/reports/summary.zip'
+detail_xml_url_template = \
+    clarity_url + '{administrator}/{election_id}/{version_num}/reports/detailxml.zip'
+county_detail_xml_url = clarity_url + \
+    '{state}/{county_name}/{county_election_id}/{county_version_num}/reports/detailxml.zip'
+detail_settings_json_url_template = \
+    clarity_url + '{state}/{election_id}/{version_num}/json/en/electionsettings.json'
+
+tzinfos = {
+    "EST": "UTC-5",
+    "EDT": "UTC-4",
+    "CST": "UTC-6",
+    "CDT": "UTC-5",
+    "MST": "UTC-7",
+    "MDT": "UTC-6",
+    "PST": "UTC-8",
+    "PDT": "UTC-7",
+    "AKST": "UTC-9",
+    "AKDT": "UTC-8",
+    "HST": "UTC-10",
+    "HDT": "UTC-9",
+}
+
+
+class CountyDetails:
+    def __init__(
+        self,
+        state: str,
+        county_name: str,
+        county_election_id: str,
+        county_version_num: str,
+        county_update_date: datetime = None
+    ):
+        self.state = state
+        self.county_name = county_name
+        self.county_election_id = county_election_id
+        self.county_version_num = county_version_num
+        self.county_update_date = county_update_date
+
+
+class Scytl(object):
+    """
+    Instantiate a Scytl connector.
+
+    `Args:`:
+        state: str
+            The two letter code of the state the publishing election results.
+            ex: GA
+        election_id: str
+            The numeric identifier for the election found in the url of the election's website.
+            ex: "114729"
+        county: str (optional)
+            The name of the county publishing the results.
+            ex: Clarke
+    """
+
+    def __init__(self, state: str, election_id: str, county=""):
```

Comment:
It's unusual for anything other than auth info to get passed in to a Parsons connector, although it certainly does happen, and it makes sense to pass in `state` and `election_id` here so they don't have to be passed in with `get_summary_results`, etc. 

That said, `county` in particular confuses me. How does the county parameter here fit with the counties passed in with `get_detailed_results_for_participating_counties`? Does this county parameter get used at any other point? It seems like the other two are always state level but maybe I'm misunderstanding. Does passing in the county on init mean that `get_summary_results` and `get_county_results` apply only to the county?

Reply:
Yeah, the setup is kinda confusing, would love your help navigating it. 

Essentially, they have a concept of an election (at the state or county level), which can have summary results and detailed results. The detailed results show results by county for state level elections, while they show precinct results for a county-level election.

State-level elections can also have sub-elections in each county, and link to each of those county elections, which each have summary and detailed results. So, a county-level election won't have participating counties and the method won't make sense.

The main reason I wanted to add the state and election_id at the class level instead of the method is so the class could track the previous version- I have a feeling people would use this for polling and I want to make that as easy as possible for them to see if results have updated.

One idea might be to split this up into a `StateScytlElection` and `CountyScytlElection` class, which would have some of the same implementation, but not all of it. What do you think?

Reply:
Sorry, that was a bit of a word salad. Does it make sense to you to split up the state-wide election and county-wide election into two separate connectors, since they have slightly different functions and outputs?

Separately, if we're using this for polling, to prevent fetching unnecessary results, does it make sense to you to store the previous version in the class or return it with the results and have them pass it back in the function?